File: apps/admin_mobile/lib/application/use_cases/reservations/get_reservations_use_case.dart
import '../../../domain/entities/reservation.dart';
import '../../../domain/repositories/reservation_repository.dart';
import '../../../shared/types/paginated_result.dart';

class GetReservationsUseCase {
  final ReservationRepository _repo;
  GetReservationsUseCase(this._repo);

  Future<PaginatedResult<Reservation>> call(ReservationFilters filters) =>
      _repo.getAll(filters);
}

File: apps/admin-v2/src/application/use-cases/reservations/get-reservations.use-case.ts
import type { ReservationRepository } from '@/domain/repositories/reservation.repository';
import type { ReservationFilters } from '@/domain/entities/reservation';

export class GetReservationsUseCase {
  constructor(private repo: ReservationRepository) {}

  execute(filters: ReservationFilters) {
    return this.repo.getAll(filters);
  }
}
